packages/slonik/src/factories/createConnectionPool.ts

```diff
@@ -144,22 +144,25 @@ export const createConnectionPool = ({
       pendingConnections.push(pendingConnection);
 
       const connection = await pendingConnection.catch((error) => {
-        pendingConnections.pop();
```

Comment:
`pendingConnections.pop();` will remove the most recently added pending connection from the array. But by the time this code runs the last connection in the array might not be the same one that threw an error right?

E.g.
- Pending Connection A added. `pendingConnections = [A]`
- Pending Connection B added. `pendingConnections = [A,B]`
- 5s later A has timed out and throws an error. `pendingConnections.pop()` is called and now B is removed and we’re left with `[A]`. But it should've been `A` that was removed